calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -305,3 +306,118 @@ def test_validate_arguments(self, m_sys_exit):
             else:
                 assert m_sys_exit.called
             m_sys_exit.reset_mock()
+
+
+class TestUtils(unittest.TestCase):
+
+    @patch('sys.exit', autospec=True)
+    def test_check_ip_version(self, m_sys_exit):
+        """
+        Test check_ip_version function in calicoctl utils
+        """
+        # Set up arguments
+        cases = (
+            ('1.2.3.4', 'v4', IPAddress, True, 1),
+            ('1.2.3.4', 'v6', IPAddress, False, 2),
+            ('1.2.3.4', 'v4', IPNetwork, True, 3),
+            ('1.2.3.0/24', 'v4', IPAddress, False, 4),
+            ('1.2.3.0/24', 'v4', IPNetwork, True, 5),
+            ('aa:bb::ff', 'v4', IPAddress, False, 6),
+            ('aa:bb::ff', 'v6', IPAddress, True, 7)
```

Comment:
These use cases look good.  I think it would be useful to test a couple more cases too:
- full IPv6 address - 1111:2222:3333:4444:5555:6666:7777:8888
- number below 4294967295 (highest possibly ipv4 address as a number)
- number above 4294967296 (parsed as an ipv6 address, since it's larger than all ipv4 addresses)